ACFIVE-32665

slider1337 opened the pull request:
REST: PSR-Handler in Zeus richtig einbauen

- correct handling of psr request
- set way for implementing middleware etc.
- be more psr compatible

simo-benedikt-schaller committed:
ACFIVE-32665
REST: PSR-Handler in Zeus richtig einbauen

- correct handling of psr request
- set way for implementing middleware etc.
- be more psr compatible

simo-benedikt-schaller committed:
ACFIVE-32665
REST: PSR-Handler in Zeus richtig einbauen

- remove type hint, because not backwards compatible

simo-benedikt-schaller committed:
ACFIVE-32665
REST: PSR-Handler in Zeus richtig einbauen

- remove type hint, because not backwards compatible

simo-benedikt-schaller committed:
ACFIVE-32665
REST: PSR-Handler in Zeus richtig einbauen

- move the RouteNotFoundException to the kernel and throw no error while routing, to ensure that psr routes are handled

Carpenter0100 commented on src/Command/CommandDispatcher.php:
```diff
@@ -27,66 +21,56 @@ class CommandDispatcher implements HttpRequestDispatcherInterface {
+            $validationResult = $commandValidator->validate($commandRequest);
+        }
+
+        if (null === $validationResult || $validationResult->isValid()) {
```
die inline kommentare sind hier nicht nötig. 
es wäre gut wenn du das if else konstrukt hier ein bisschen aufräumst. Also nach dem Prinzip so schnell wie möglich aus der Methode raus.

Carpenter0100 commented on src/Handler/HandlerRouterChain.php:
```diff
@@ -51,6 +49,6 @@ public function route(ServerRequestInterface $request): RequestHandlerInterface
         }
         
-        throw new RouteNotFoundException(new Url($request->getUri()->getPath()));
+        return null;
```
warum return null? Könnte man das nicht drin lassen und die Exception abfangen und dann die router chain weiter durchlaufen?

slider1337 replied:
Macht für mich keinen Sinn bei allen drei chains ein try catch drum rum zu basteln, aber ich pack das wie besprochen in den router

Carpenter0100 commented on src/Http/HttpKernel.php:
```diff
@@ -125,21 +127,22 @@ public function run(HttpRequestInterface $request, $send = true)
+            $psrRequest = $this->createPsrRequestFromZeusRequest($request);
+            $route = $router->route($request, $psrRequest);
+            if ($route === null) {
+                throw new RouteNotFoundException($request->getUrl());
```
Der Dispatcher sollte diese Exception nicht werfen. Das ist teil des routings.

Carpenter0100 commented on src/Http/HttpKernel.php:
```diff
@@ -196,4 +200,10 @@ abstract protected function registerBundles(HttpRequestInterface $request);
+
+    private function createPsrRequestFromZeusRequest(HttpRequestInterface $request): ServerRequestInterface
+    {
+        $psrRequest = ServerRequest::fromGlobals();
```
Nutz hier am besten ne ServerRequestFactory

Carpenter0100 commented on src/Http/Request/HttpRequestDispatcherLocator.php:
```diff
@@ -22,22 +22,20 @@ class HttpRequestDispatcherLocator {
+        if ($route->isGetRoute()) {
+            return $this->frameworkFactory->createHttpGetRequestDispatcher();
+        }
+        if ($route->isPsrRoute()) {
```
kann eine psr route nicht auch ein get sein? Oder wie ist die Unterscheidung hier?

slider1337 replied:
N des betrifft die alten zeus routen die waren unterteilt auf get und post, psr kann alles sein

Carpenter0100 commented on src/Http/Routing/HttpRouter.php:
```diff
@@ -0,0 +1,87 @@
+     * @param UriInterface $uri
+     * @return HandlerRouteParameterMapInterface
+     */
+    private function createRouteParameterMap(UriInterface $uri): HandlerRouteParameterMapInterface {
```
PSR

Carpenter0100 commented on src/Http/Routing/HttpRouter.php:
```diff
@@ -0,0 +1,87 @@
+            return $this->routeFactory->createPsrRoute($psrRequest, $psrHandler);
+        }
+
+        // Route not found
```
Warum keine Exception?

Carpenter0100 reviewed (changes requested):
Also, alles in allem finde ich die Umsetzung nicht gut gelungen. Ich hätte es vermutlich ganz anders gemacht.
Können wir später nochmal drüber reden.

simo-benedikt-schaller committed:
ACFIVE-32665
REST: PSR-Handler in Zeus richtig einbauen

- some psr formatting of old files
- move creation of server request to own factory, in that way we can replace it easier later
- Throw route not found exception in router, because of cr

simo-benedikt-schaller committed:
ACFIVE-32665
REST: PSR-Handler in Zeus richtig einbauen

- delete unused files
